### DISCUSSION TITLE
IFRS9 Consecutive Days on Default

### DISCUSSION THREAD

**Discussion by alexis-bvq** [atoti/atoti#625] [General]
## Description
We have a measure that count the number of consecutive day where clients didn't pay their credit card on time. Therefore we put the scope cumulative over this date hierarchy. The issue is that if for few days he started paying on time again, but then months later start to pay after due date, the cumulative doesn't not restart to 0.

![image](https://user-images.githubusercontent.com/47662440/95057163-ad555a80-06ed-11eb-9c97-6dd2db493ded.png)

This is what you can see on this picture. He started to be on default the first two months (until 2018-10-31). Then from 2019-01-31 to 2019-03-31 he paid on time. Finally on the 2019-04-30 he started to be on default again. Nevertheless, the consecutive days on default should start again to 1.

We need an option on the cumulative scope to restart the accumulation once you hit a null on the previous member.

## Context
This is for the IFRS9 demo.

**Comment by rcolle** [comment]
Can you give a bit more details about the measures involved and their values? I can see there are no values in the second widget before 2018-11-30, but how can you tell that he's late on 2019-04-30?

One issue I see here is that, ActivePivot-wise, a `null` value means "non-existing" data. Therefore a measure is not evaluated on coordinates where all its underlying values are `null`.

**Comment by alexis-bvq** [comment]
The data is about credit risk. We want to monitor if the clients pay on time and the risk link to that. In the data set for each client, for each month, you know the number of `Days past due`. It represents the number of days after his billing date that the bank had to wait for him to pay. Most of the time it is 0 but some clients have some issues to pay their credit on time and they are late on payments. 

On this data set we have data for each end of the month. We want a measure to know how many consecutive months this amount was not 0. It represents the number of consecutive month the client was late on his payments. This measure should be reset to 0 once there is a month when he paid on time. 

The issue with the cumulative scope is that this amount is not reset if you have a null value as you commented before. Nevertheless, there is an option to fill `null` locations with the cumulated value `dense=True`. If there could be an option to have for example `reset=True` that would be perfect.
